### DISCUSSION TITLE
Accessing Strimzi kafka outside of Kubernetes using Gloo

### DISCUSSION THREAD

**Discussion by rishinair19** [strimzi/strimzi-kafka-operator#6299] [General]
We've been using Strimzi for Kafka in AWS but now we're migrating to DC. For listener, we were using Loadbalancer till now and it was working well with AWS. 

But now we're moving to DC so we can't really use any LBs. We don't wish to go with Nodeport as its not stable. With ingress, we can't use SSL so basically we can't use any of the 3 external listeners.

For exposure, we do have Gloo and I did try to expose the kafka bootstrap service but I'm unable to connect to it over port 9092. It seems connection is being blocked but I'm not really sure. I wanted to confirm is this because I have not configured an external listener in Kafka config?

**Comment by scholzj** [comment]
I do not know Gloo hands-on. But one thing to keep in mind is that Kafka is a TCP connection. So common issue with API Gateways would be how they deal with TCP -> especially if you don't want to use TLS and some form of TLS passthrough with Gloo either (since you do not want to use it with Ingress, I guess it might be the same?)

**Reply by rishinair19** [reply]
Thank you for replying. I understand that can be an issue.

I just wanted to confirm that if we don't specify an external listener in Kafka configuration and then expose the bootstrap service directly, will that work?

**Reply by scholzj** [reply]
No, it is not so easy. First of all, you would need to expose all brokers and not just the bootstrap service. Second, you would also need to configure the advertised hosts to match the addresses on your gateway. Check out this blog post series: https://strimzi.io/blog/2019/04/17/accessing-kafka-part-1/ ... tht should give you better idea how Kafka works with regards to this.
